packages/dscc-scripts/src/viz/util.ts

```diff
@@ -118,3 +120,66 @@ export const validateConfigFile = (path: PathLike): boolean => {
   throwIfErrors(validate.validateConfig(parsedJson), 'config');
   return true;
 };
+
+export const getBuildableComponents = (): ComponentBuildValues[] => {
+  const components: ComponentBuildValues[] = [];
+
+  const lastComponentIdx = Object.keys(process.env)
+    .filter((key) => key.startsWith('npm_package_dsccViz_components_'))
+    .map((s) => s.replace('npm_package_dsccViz_components_', ''))
+    .map((a) => parseInt(a, 10))
+    .reduce((a, b) => (a > b ? a : b), 0);
+
+  // Check for vizpack configuration
+  for (let idx = 0; idx <= lastComponentIdx; idx++) {
+    const jsonFile =
+      process.env[`npm_package_dsccViz_components_${idx}_jsonFile`];
+
+    if (!jsonFile) {
+      throw invalidVizConfig(`components[${idx}].jsonFile`);
+    }
+
+    const cssFile =
+      process.env[`npm_package_dsccViz_components_${idx}_cssFile`];
+    // Require either jsFile or tsFile
+    const jsFile = process.env[`npm_package_dsccViz_components_${idx}_jsFile`];
+    const tsFile = process.env[`npm_package_dsccViz_components_${idx}_tsFile`];
+
+    if (jsFile === undefined && tsFile === undefined) {
+      if (jsFile === undefined) {
+        throw invalidVizConfig(`components[${idx}].jsFile`);
+      }
+      if (tsFile === undefined) {
```

Comment:
Technically you'd never end up in this case because of the guarding if statement. I think you can just remove it, we don't show that you can use TS very well anyway.

Reply:
Oh good catch. I thought the outer guard was an or. Update.